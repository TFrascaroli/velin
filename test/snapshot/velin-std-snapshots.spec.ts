import { describe, it, expect } from 'vitest'
import Velin from '../../src/velin-core.js'
import setupVelinStd from '../../src/velin-std.js';
setupVelinStd(Velin);


describe('bind', () => {
  it('renders bound text', () => {
    const div = document.createElement('div')
    div.setAttribute('vln-text', 'vln.message')

    Velin.bind(div, { message: 'hello world' })

    expect(div).toMatchSnapshot()
  });

  it('renders loop', () => {
    const div = document.createElement('div')
    div.setAttribute('vln-text', 'vln.message')

    Velin.bind(div, { message: 'hello world' })

    expect(div).toMatchSnapshot()
  });
});
