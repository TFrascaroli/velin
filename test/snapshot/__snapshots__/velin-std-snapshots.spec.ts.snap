// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bind > renders bound text 1`] = `
<div
  reflect-vln-text="vln.message"
>
  hello world
</div>
`;

exports[`bind > renders loop 1`] = `
<div
  reflect-vln-text="vln.message"
>
  hello world
</div>
`;
